### PR TITLE
Test circuit recovery against failure frequency

### PR DESCRIPTION
**Pull request by kriskowal**

### CODE REVIEW

File: node/test/circuits.js [hunk 1]
```diff
@@ -417,6 +417,74 @@ RelayNetwork.test('recovers after five successes', aliceAndBob, function t(netwo
...
 
+RelayNetwork.test('does not recover when successes are non-consecutive', aliceAndBob, function t(network, assert) {
+
+    var count = 0;
+
+    var circuit = network.getCircuit(0, 'alice', 'bob', 'call');
+
+    network.register('call', function (req, res) {
+        count++;
+        if (count > 20 && count % 4 !== 0) {
```
Comment (kriskowal) [inline node/test/circuits.js]: Note that in this case, 1 in 4 fail after the circuit is forced into the unhealthy state.

File: node/test/circuits.js [hunk 2]
```diff
@@ -417,6 +417,74 @@ RelayNetwork.test('recovers after five successes', aliceAndBob, function t(netwo
...
+
+RelayNetwork.test('recovers when failure is periodic but infrequent', aliceAndBob, function t(network, assert) {
+
+    var count = 0;
+
+    var circuit = network.getCircuit(0, 'alice', 'bob', 'call');
+
+    network.register('call', function (req, res) {
+        count++;
+        if (count > 20 && count % 8 !== 0) {
```
Comment (kriskowal) [inline node/test/circuits.js]: Whereas in this case, one in 8. There is some interplay with the number of requests that occur per period, as approximately determined by the exercise count and delay.

**Comment by kriskowal** [comment]
r @jcorbin @Raynos @breerly

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**